From 955b7f57b2ece3d8aa1af145a59ca694b4633c0f Mon Sep 17 00:00:00 2001
From: Zheng Yang <zhengyang@rock-chips.com>
Date: Tue, 8 Dec 2015 08:46:06 +0800
Subject: [PATCH] video: rockchip: fb: disable mutex when load_screen

This mutex is conflict witch mutex located at rk_fb_update_reg,
it can make system deadlock and showing black picture.

Change-Id: I55192e8cd32961ef24dd950639055bbf56e9fec9
Signed-off-by: Zheng Yang <zhengyang@rock-chips.com>
---
 drivers/video/rockchip/rk_fb.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index 976089450d53..69e6e852308a 100644
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -3535,9 +3535,7 @@ int rk_fb_switch_screen(struct rk_screen *screen, int enable, int lcdc_id)
 			win_id = dev_drv->ops->fb_get_win_id(dev_drv, info->fix.id);
 			win = dev_drv->win[win_id];
 			if (win && fb_par->state) {
-				mutex_lock(&dev_drv->win_config);
 				dev_drv->ops->load_screen(dev_drv, 1);
-				mutex_unlock(&dev_drv->win_config);
 
 				info->var.activate |= FB_ACTIVATE_FORCE;
 				if (rk_fb->disp_mode == ONE_DUAL) {
-- 
2.35.3

